src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -631,20 +635,22 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
             "JITCompilationStarted skipping method: Method replacement "
             "found but the managed profiler has not yet been loaded. "
             "function_id=",
-            function_id, " token=", function_token, " name=", caller.type.name,
-            ".", caller.name, "()");
+            function_id, " token=", function_token, " target_name=", target.type.name,
+            ".", target.name, "()");
         continue;
       }
 #endif
 
       const auto original_argument = pInstr->m_Arg32;
+      const void* module_version_id_ptr = &module_metadata->module_version_id;
 
       // insert the opcode and signature token as
       // additional arguments for the wrapper method
       ILRewriterWrapper rewriter_wrapper(&rewriter);
       rewriter_wrapper.SetILPosition(pInstr);
       rewriter_wrapper.LoadInt32(pInstr->m_opcode);
       rewriter_wrapper.LoadInt32(method_def_md_token);
+      rewriter_wrapper.LoadInt64(reinterpret_cast<INT64>(module_version_id_ptr));
```

Comment:
We treat the address of `module_version_id` as an `INT64` and we push it into the stack before calling the wrapper method.